blueman/Functions.py

```diff
@@ -112,18 +110,14 @@ def launch(
     icon_name: Optional[str] = None,
     name: str = "blueman",
     sn: bool = True,
+    launched_cb: Optional[Callable[[Gio.AppLaunchContext, Gio.AppInfo, GLib.Variant], None]] = None
 ) -> bool:
     """Launch a gui app with starup notification"""
-    context = None
-    gtktimestamp = Gtk.get_current_event_time()
-    if gtktimestamp == 0:
-        logging.info("Gtk eventtime is 0, not using LaunchContext")
-        timestamp = int(time.clock_gettime(time.CLOCK_MONOTONIC_RAW))
-    else:
-        timestamp = gtktimestamp
-        display = Gdk.Display.get_default()
-        context = display.get_app_launch_context()
-        context.set_timestamp(timestamp)
+
+    timestamp = Gtk.get_current_event_time()
+    display = Gdk.Display.get_default()
+    context = display.get_app_launch_context()
+    context.set_timestamp(timestamp)
```

Comment:
Don't undo this, see #1269 for details

Reply:
I see, I have adjusted the code to make up a timestamp when GTK can't provide one. I have tested the behavior from #1269 on MATE 1.24 and it seems to work fine.